lint/spotbugs.bzl

```diff
@@ -0,0 +1,142 @@
+    inputs = srcs + [exclude_filter]
+    outputs = [stdout]
+    args = ctx.actions.args()
+    args.add_all(options)
+
+    if exclude_filter:
+        args.add_all(["-exclude", exclude_filter.path])
+
+    src_args = ctx.actions.args()
+    src_args.add_all(srcs)
+    args.add_all(["-exclude", exclude_filter.path])
+
+    if exit_code:
+        command = "{SPOTBUGS} $@ >{stdout}; echo $? > " + exit_code.path
+        outputs.append(exit_code)
+    else:
+        # Create empty stdout file on success, as Bazel expects one
+        command = "{SPOTBUGS} $@ && touch {stdout}"
+    ctx.actions.run_shell(
+        inputs = inputs,
+        outputs = outputs,
+        command = command.format(SPOTBUGS = executable.path, stdout = stdout.path),
+        arguments = [args, src_args],
+        mnemonic = _MNEMONIC,
+        tools = [executable],
+        progress_message = "Linting %{label} with Spotbugs",
+    )
+
+# buildifier: disable=function-docstrings
+def _spotbugs_aspect_impl(target, ctx):
+    if not should_visit(ctx.rule, ctx.attr._rule_kinds):
+        return []
+
+    print("Print target %s", target.label)
+    print("JavaInfo %s", target[JavaInfo])
+    print("OutputGroupInfo %s", target[OutputGroupInfo])
+
+    files_to_lint = [jar.class_jar for jar in target[JavaInfo].outputs.jars]
+    print("Files to lint %s", files_to_lint)
+    outputs, info = output_files(_MNEMONIC, target, ctx)
+    if len(files_to_lint) == 0:
+        noop_lint_action(ctx, outputs)
+        return [info]
+    format_options = []  # to define
+    spotbugs_action(ctx, ctx.executable._spotbugs, files_to_lint, ctx.file._exclude_filter, outputs.human.out, outputs.human.exit_code, format_options)
+    spotbugs_action(ctx, ctx.executable._spotbugs, files_to_lint, ctx.file._exclude_filter, outputs.machine.out, outputs.machine.exit_code, format_options)
+    return [info]
+
+def lint_spotbugs_aspect(binary, exclude_filter, rule_kinds = ["java_library", "java_binary"]):
+    return aspect(
+        implementation = _spotbugs_aspect_impl,
+        # Edges we need to walk up the graph from the selected targets.
+        # Needed for linters that need semantic information like transitive type declarations.
+        attr_aspects = ["deps"],
+        attrs = {
+            "_options": attr.label(
+                default = "//lint:options",
+                providers = [LintOptionsInfo],
+            ),
+            "_spotbugs": attr.label(
+                default = binary,
+                executable = True,
+                cfg = "exec",
+            ),
+            "_exclude_filter": attr.label(
+                doc = "Report all bug instances except those matching the filter specified by this filter file",
+                allow_single_file = True,
+                default = exclude_filter,
+            ),
+            "_rule_kinds": attr.string_list(
+                default = rule_kinds,
+            ),
+        },
+    )
+
+def fetch_spotbugs():
+    http_archive(
```

Comment:
we should have something to keep the version here updated, and also make it bzlmod-friendly.

But since you're copying an existing pattern in the repo we can merge this as-is and improve later :)